### DISCUSSION TITLE
Not ideal HistogramBucket for durationHistogram

### DISCUSSION THREAD

**Discussion by dieend** [fluxcd/flux2#1870] [Q&A]
In [here](https://github.com/fluxcd/pkg/blob/main/runtime/metrics/recorder.go#L62), we bucketed everything that's longer than 10s into [10, +inf] bucket metrics. This is really not ideal, as seen from this chart
![image](https://user-images.githubusercontent.com/1404443/134973327-b98da26f-d7b9-4f78-b685-c10b8b06c565.png)

Are we assuming that every reconciliation should be completed under 10s? We are assigning the following limits to the kustomize-controller with concurrency = 24.
```
    cpu: 6000m
    memory: 6Gi
```

1. Are we encountering a performance bottleneck? How do we scale it up?
2. Can we use custom histogram buckets?

**Comment by kingdonb** [comment]
I see there is a PR open to address this issue in fluxcd/pkg:

* https://github.com/fluxcd/pkg/pull/164
(thank you @dieend !)
